packages/block-editor/src/components/block-controls/fill.js

```diff
@@ -21,14 +22,24 @@ export default function BlockControlsFill( {
 		group,
 		__experimentalShareWithChildBlocks
 	);
+	const blockEditingMode = useBlockEditingMode();
+
 	if ( ! Fill ) {
 		return null;
 	}
 
+	// Filter children in content-only mode
+	let filteredChildren = children;
+	if ( blockEditingMode === 'contentOnly' && Array.isArray( children ) ) {
+		filteredChildren = children.filter( ( child ) => {
```

Comment:
This `filter` will always create a new `Array`. For perf reasons I wonder if there's a way to avoid that 🤔

Reply:
We can certainly look to memoize.